Source/Core/Core/DSP/DSPHWInterface.cpp

```diff
@@ -227,20 +230,24 @@ u16 gdsp_ifx_read(u16 addr)
 
 static const u8* gdsp_idma_in(u16 dsp_addr, u32 addr, u32 size)
 {
+  const u8* code = &g_dsp.cpu_ram[addr & 0x0fffffff];
+  g_dsp.iram_crc = HashEctor(code, size);
+
+  if (SConfig::GetInstance().m_DumpUCode)
+    DumpDSPCode(code, size, g_dsp.iram_crc);
+
   Common::UnWriteProtectMemory(g_dsp.iram, DSP_IRAM_BYTE_SIZE, false);
 
   u8* dst = ((u8*)g_dsp.iram);
   for (u32 i = 0; i < size; i += 2)
   {
-    *(u16*)&dst[dsp_addr + i] =
-        Common::swap16(*(const u16*)&g_dsp.cpu_ram[(addr + i) & 0x0fffffff]);
+    *(u16*)&dst[dsp_addr + i] = Common::swap16(*(const u16*)&code[i]);
```

Comment:
This is kinda undefined behavior; this should probably use `memcpy`/`memmove` instead (technically, both before and after the equal sign). You can also look at the overload of `Common::swap16` that takes a pointer (vs. the one you're calling right now that takes a `u16`).
Should be fixed at some point, and might as well do it now that you're touching the code anyways.